src_new/adapterEntry.js

```diff
@@ -24,6 +24,9 @@ AdapterEntry.prototype.getBid = function(bidID){
 };
 
 AdapterEntry.prototype.setNewBid = function(theBid){
+	if (theBid.kgpv === "") {
+		theBid.kgpv = this.bids[this.lastBidID].kgpv;
```

Comment:
This code will return the wrong kgpv in case of multi-size slots.
Say kgp is _AU_@_W_x_H_, old bid is of size 300x250 and new bid is of size 300x300.

Reply:
The case that I am trying to fix is as follows:
- when hbopenbid returns an errorCode like zero bids/time out etc (anything in error code 1-6), we get width and height as 0 in the bid object.  In that case, kgpv was getting set to an incorrect value and hence logged incorrectly. The original bid object created from the adslot config has the correct kgpv set. Hence using the kgpv from the original bid object before its overwritten by the new one (which does not have the correct kgpv) Refer to line https://github.com/PubMatic/OpenWrap/pull/55/files#diff-9738689368368781d93015f9a6bbf233R30. 
The case you have mentioned will arise if a valid bid is returned. And in that case, theBid.kgpv will not be blank.
Let me know if this makes sense

Reply:
I understand the problem you are trying to solve,
But this proposed solution is not scalable and is error-prone. If you are submitting multiple error bids for a slot then this code will not set the right KGPV.